sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/NumberConverter.scala

```diff
@@ -49,12 +49,14 @@ object NumberConverter {
    */
   private def encode(radix: Int, fromPos: Int, value: Array[Byte]): Long = {
     var v: Long = 0L
-    val bound = java.lang.Long.divideUnsigned(-1 - radix, radix) // Possible overflow once
+    // bound will always positive since radix >= 2
+    val bound = java.lang.Long.divideUnsigned(-1 - radix, radix)
```

Comment:
I'm not very familiar with this code. What does `bound` mean here?

Reply:
bound is the max value which can multiply by radix without overflow when we treat Long value as Unsigned Long value

Reply:
so if v is greater than bound, v*radix will cause overflow

Reply:
note that when treat Long value as unsigned long value, as negative value will start with bit 1, negative value will be greater than positive values.
the presentation of unsigned long type is from 0000000000000000000000000000000000000000000000000000000000000000 (equal to 0 in long type)
to 
1111111111111111111111111111111111111111111111111111111111111111 (equal to -1 in long type)